### DISCUSSION TITLE
How to update multiple forms

### DISCUSSION THREAD

**Discussion by moazam1** [livewire/livewire#4050] [1. Help] [answered]
I am displaying users list on the manage page along with several other actions i.e. update account status, etc. I have created a modal to allow admins to update individual user status. To do that I have created a form in the modal and it has select and save button. Here is the form excerpt:

```
@foreach ($users as $user)
  <form wire:submit.prevent="updateAccountStatus">
     <input type="hidden" name="user_id" value="{{ $user->_id }}" wire:model="userId">
     <select class="select2 form-control" wire:model="accountStatus">
          <option  value="active" {{ $user->account_status == 'active' ? 'selected' : '' }}>Active</option>
            .....
       </select>
  <button type="submit" class="btn btn-primary">Update</button>
  </form>
@endforeach
```

Code from my component:

   public $userId;
    public $accountStatus;

    public function updateAccountStatus( ) {
        log_this($this->accountStatus);
    }

`$this->accountStatus` shows me null unless I change the value and once the value is changed it shows same selected value for all other users which obviously makes sense because I am using public property. I don't have much livewire knowledge so asking community to put me in the right direction. Thank you

**Comment by joshhanley** [comment]
@moazam1 you could create a Livewire component for your account status form, so each user will have it's own Livewire component, so when you click "Update" it will only update that user.

So it would look like 

```blade
@foreach ($users as $user)
  <livewire:account-status-form :user="$user" wire:key="account-status-form-{{ $user->id }}" />
@endforeach
```

Then move your form html into that component and move your method and properties into it too.

Hope this helps!

**Reply by moazam1** [reply]
@joshhanley thank you for the reply. I ended up doing same thing but now I am having another issue. When I update the status, parent component wasn't updating so I did something like this in the child component update method:

$this->user->save();
`$this->emit('userUpdated');`

And in the parent component I have a listener:

`protected $listeners = ['userUpdated' => '$refresh'];`

The issue with above approach is, it only updates parent component field once and when I update it again it doesn't seem to do anything. If I am not able to figure out solution shortly, I may try this one: https://www.youtube.com/watch?v=RdSYT8Hivsg bit of Javascript but it works in my situation.

**Comment by moazam1** [comment, marked as answer]
Closing this request as I was able to solve using the method suggested in the above video.
